cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/DynamicDatasetCache.java

```diff
@@ -191,6 +191,12 @@ public DynamicDatasetCache(SystemDatasetInstantiator instantiator,
   public final <T extends Dataset> T getDataset(String namespace, String name, Map<String, String> arguments,
                                                 boolean bypass, AccessType accessType)
     throws DatasetInstantiationException {
+    if (namespace.equals(NamespaceId.SYSTEM.getNamespace()) &&
+      !this.namespace.getNamespace().equals(NamespaceId.SYSTEM.getNamespace())) {
```

Comment:
shouldn't need to call `getNamespace()` to compare. Should just be `NamespaceId.SYSTEM.equals(this.namespace)`